mlir/lib/Dialect/Rock/utility/builderUtils.cpp

```diff
@@ -86,7 +86,7 @@ Value createZeroConstantOp(OpBuilder &b, Location loc, Type type) {
   if (elementType.isIntOrIndex()) {
     return createConstantIntOp(b, loc, type, elementType, 0);
   } else {
-    return createConstantFloatOp(b, loc, type, elementType, 0.0);
+    return createConstantFloatOp(b, loc, type, elementType, 0.0f);
```

Comment:
shouldn't be necessary to add `f` at the end since it would bind with argument of type `float`.